shepherd.js/package.json

```diff
@@ -18,28 +18,40 @@
     "Robbie Wagner <rwwagner90@gmail.com>",
     "Chuck Carpenter <chuck@shipshape.io>"
   ],
-  "main": "dist/shepherd.js",
-  "module": "dist/shepherd.mjs",
+  "main": "./dist/cjs/shepherd.cjs",
+  "module": "./dist/esm/shepherd.mjs",
   "exports": {
     ".": {
-      "import": "./dist/shepherd.mjs",
-      "require": "./dist/shepherd.js"
+      "require": {
+        "types": "./dist/cjs/shepherd.d.cts",
+        "default": "./dist/cjs/shepherd.cjs"
+      },
+      "import": {
+        "types": "./dist/esm/shepherd.d.ts",
+        "default": "./dist/esm/shepherd.mjs"
+      }
     },
     "./*": {
-      "import": "./dist/*.mjs",
-      "require": "./dist/*.js"
+      "require": {
+        "types": "./dist/cjs/*.d.cts",
+        "default": "./dist/cjs/*.cjs"
+      },
+      "import": {
+        "types": "./dist/esm/*.d.ts",
+        "default": "./dist/esm/*.mjs"
+      }
     }
   },
-  "types": "shepherd.d.ts",
+  "type": "module",
   "typesVersions": {
     "*": {
       "*": [
-        "./dist/*"
+        "./dist/esm/*"
```

Comment:
The `moduleResolution: Node10` issues flagged by attw are caused by typesVersions, which is only needed to make subpath exports work with `moduleResolution: Node10`. `tsc --traceResolution` with a sample project shows that this is failing because TypeScript takes the `*` as a literal substitution - it tries to resolve `dist/cjs/shepherd.cjs` from `main`, substitutes it in the `*`, and gets `dist/esm/dist/cjs/shepherd.cjs`.

If you add a specific mapping for shepherd.cjs, then `moduleResolution: node10` seems to work:

```
      "dist/cjs/shepherd.cjs": [
        "./dist/cjs/shepherd.d.cts"
      ],
      "*": [
        "./dist/cjs/*"
      ]
```

However, I'm not sure that you need typesVersions - see comments elsehwere.

Reply:
Removing `typesVersions` or changing it with what you suggest here are both fine with me. I had deleted `typesVersions` a few times and it didn't seem to change attw at all whether I had it or not.